### ISSUE TITLE
Deploy failed on Netlify

### ISSUE THREAD

**Issue by omrllm** [gatsbyjs/gatsby#10626]
## Description

I get a deploy failed message when I try to deploy

### Steps to reproduce

This is the repo that I'm trying to deploy on Netlify: https://github.com/omrllm/Getsy

- New site from Git [on the Netlify panel]
- Continuous deployment - Github
- Select the Getsy repo 

**Branch to deploy: master
Build command: gatsby build
Publish directory: public/ or public (both give me an error)**

### Expected result

Successfully deploy the my website

### Actual result

Deploy failed
Production: master@HEAD

**Deploy log**

```
6:32:55 PM: Build ready to start
6:32:57 PM: build-image version: 53a2d5fe9ca22a80747972bf7a83d93509ec503c
6:32:57 PM: buildbot version: 4011bc275ed1e45f93e758dff0bce4fda2a3c46f
6:32:57 PM: Fetching cached dependencies
6:32:57 PM: Failed to fetch cache, continuing with build
6:32:58 PM: Starting to prepare the repo for build
6:32:59 PM: No cached dependencies found. Cloning fresh repo
6:32:59 PM: git clone https://github.com/omrllm/Getsy
6:32:59 PM: Preparing Git Reference refs/heads/master
6:33:00 PM: Starting build script
6:33:00 PM: Installing dependencies
6:33:01 PM: Downloading and installing node v8.14.1...
6:33:01 PM: Downloading https://nodejs.org/dist/v8.14.1/node-v8.14.1-linux-x64.tar.xz...
6:33:01 PM:   0.
6:33:01 PM: 0%
6:33:01 PM: #
6:33:01 PM: ###                                                                       5.9%
6:33:01 PM: #
6:33:01 PM: #########################################                                59.3%
6:33:01 PM: #
6:33:01 PM: ####################################################################### 100.0%
6:33:01 PM: Computing checksum with sha256sum
6:33:02 PM: Checksums matched!
6:33:04 PM: Now using node v8.14.1 (npm v6.4.1)
6:33:04 PM: Attempting ruby version 2.3.6, read from environment
6:33:06 PM: Using ruby version 2.3.6
6:33:06 PM: Using PHP version 5.6
6:33:06 PM: Started restoring cached node modules
6:33:06 PM: Finished restoring cached node modules
6:33:06 PM: Installing NPM modules using NPM version 6.4.1
6:33:46 PM: > sharp@0.21.1 install /opt/build/repo/node_modules/sharp
6:33:46 PM: > (node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)
6:33:46 PM: info
6:33:46 PM:  sharp Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz
6:33:49 PM: > cwebp-bin@5.0.0 postinstall /opt/build/repo/node_modules/cwebp-bin
6:33:49 PM: > node lib/install.js
6:33:50 PM:   ✔ cwebp pre-build test passed successfully
6:33:50 PM: > mozjpeg@6.0.1 postinstall /opt/build/repo/node_modules/mozjpeg
6:33:50 PM: > node lib/install.js
6:33:50 PM:   ✔ mozjpeg pre-build test passed successfully
6:33:50 PM: > pngquant-bin@5.0.1 postinstall /opt/build/repo/node_modules/pngquant-bin
6:33:50 PM: > node lib/install.js
6:33:51 PM:   ✔ pngquant pre-build test passed successfully
6:33:54 PM: npm
6:33:54 PM: WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
6:33:54 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
6:33:54 PM: added 1817 packages from 1132 contributors and audited 23951 packages in 46.549s
6:33:54 PM: found 0 vulnerabilities
6:33:54 PM: NPM modules installed
6:33:54 PM: Started restoring cached go cache
6:33:54 PM: Finished restoring cached go cache
6:33:54 PM: unset GOOS;
6:33:54 PM: unset GOARCH;
6:33:54 PM: export GOROOT='/opt/buildhome/.gimme/versions/go1.10.linux.amd64';
6:33:54 PM: export PATH="/opt/buildhome/.gimme/versions/go1.10.linux.amd64/bin:${PATH}";
6:33:54 PM: go version >&2;
6:33:54 PM: export GIMME_ENV='/opt/buildhome/.gimme/env/go1.10.linux.amd64.env';
6:33:54 PM: go version go1.10 linux/amd64
6:33:54 PM: Installing missing commands
6:33:54 PM: Verify run directory
6:33:54 PM: Executing user command: gatsby build
6:33:56 PM: success open and validate gatsby-configs — 0.007 s
6:33:56 PM: success load plugins — 0.164 s
6:33:57 PM: success onPreInit — 0.492 s
6:33:57 PM: success delete html and css files from previous builds — 0.006 s
6:33:57 PM: success initialize cache — 0.008 s
6:33:57 PM: success copy gatsby files — 0.023 s
6:33:57 PM: success onPreBootstrap — 0.004 s
6:33:57 PM: success source and transform nodes — 0.059 s
6:33:57 PM: success building schema — 0.240 s
6:33:57 PM: success createPages — 0.000 s
6:33:57 PM: success createPagesStatefully — 0.037 s
6:33:57 PM: success onPreExtractQueries — 0.004 s
6:33:57 PM: success update schema — 0.126 s
6:33:57 PM: success extract queries from components — 0.107 s
6:33:58 PM: error UNHANDLED EXCEPTION
6:33:58 PM: 
6:33:58 PM:   TypeError: queue.off is not a function
6:33:58 PM:   
6:33:58 PM:   - page-query-runner.js:162 Queue.onDrain
6:33:58 PM:     [repo]/[gatsby]/dist/internal-plugins/query-runner/page-query-runner.js:162:    13
6:33:58 PM:   
6:33:58 PM:   - queue.js:469 Queue._drained
6:33:58 PM:     [repo]/[better-queue]/lib/queue.js:469:8
6:33:58 PM:   
6:33:58 PM:   - queue.js:505 Queue._processNextIfAllowed
6:33:58 PM:     [repo]/[better-queue]/lib/queue.js:505:14
6:33:58 PM:   
6:33:58 PM:   - queue.js:489 Timeout._onTimeout
6:33:58 PM:     [repo]/[better-queue]/lib/queue.js:489:12
6:33:58 PM:   
6:33:58 PM: 
6:33:58 PM: (sharp:1393): GLib-CRITICAL **: 17:33:58.478: g_hash_table_lookup: assertion 'hash_table != NULL' failed
6:33:58 PM: Caching artifacts
6:33:58 PM: Started saving node modules
6:33:58 PM: Finished saving node modules
6:33:58 PM: Started saving pip cache
6:33:58 PM: Finished saving pip cache
6:33:58 PM: Started saving emacs cask dependencies
6:33:58 PM: Finished saving emacs cask dependencies
6:33:58 PM: Started saving maven dependencies
6:33:58 PM: Finished saving maven dependencies
6:33:58 PM: Started saving boot dependencies
6:33:58 PM: Finished saving boot dependencies
6:33:58 PM: Started saving go dependencies
6:33:58 PM: Finished saving go dependencies
6:33:59 PM: Cached node version v8.14.1
6:33:59 PM: Error running command: Build script returned non-zero exit code: 1
6:33:59 PM: Failing build: Failed to build site
6:33:59 PM: failed during stage 'building site': Build script returned non-zero exit code: 1
6:33:59 PM: Finished processing build request in 1m1.936379885s
```

### Environment

```
System:
    OS: Linux 4.19 Fedora 29 (Workstation Edition) 29 (Workstation Edition)
    CPU: (4) x64 Intel(R) Core(TM) i5-5300U CPU @ 2.30GHz
    Shell: 4.4.23 - /bin/bash
  Binaries:
    Node: 11.3.0 - ~/.nvm/versions/node/v11.3.0/bin/node
    Yarn: 1.12.3 - /var/lib/snapd/snap/bin/yarn
    npm: 6.5.0 - ~/.nvm/versions/node/v11.3.0/bin/npm
  Browsers:
    Firefox: 64.0
  npmPackages:
    gatsby: ^2.0.53 => 2.0.74 
    gatsby-image: ^2.0.20 => 2.0.25 
    gatsby-plugin-manifest: ^2.0.9 => 2.0.12 
    gatsby-plugin-offline: ^2.0.16 => 2.0.20 
    gatsby-plugin-react-helmet: ^3.0.2 => 3.0.5 
    gatsby-plugin-sharp: ^2.0.14 => 2.0.16 
    gatsby-source-filesystem: ^2.0.8 => 2.0.12 
    gatsby-transformer-sharp: ^2.1.8 => 2.1.9 
  npmGlobalPackages:
    gatsby-cli: 2.4.7

error Couldn't find the required `xsel` binary. On Debian/Ubuntu you can install it with: sudo apt install xsel


  Error: Couldn't find the required `xsel` binary. On Debian/Ubuntu you can inst  all it with: sudo apt install xsel
  
  - envinfo.js:1 i
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:65651
  
  - envinfo.js:1 Object.copySync
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:66177
  
  - envinfo.js:1 Object.t.writeSync.e [as writeSync]
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:48777
  
  - envinfo.js:1 
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:46934
  
  - envinfo.js:1 Promise.all.then.e
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:46949
  
  - util.js:16 tryCatcher
    [lib]/[gatsby-cli]/[bluebird]/js/release/util.js:16:23
  
  - promise.js:512 Promise._settlePromiseFromHandler
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:512:31
  
  - promise.js:569 Promise._settlePromise
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:569:18
  
  - promise.js:614 Promise._settlePromise0
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:694:18
  
  - promise.js:638 Promise._fulfill
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:638:18
  
  - promise_array.js:126 PromiseArray._resolve
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise_array.js:126:19
  
  - promise_array.js:144 PromiseArray._promiseFulfilled
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise_array.js:144:14
  
  - promise.js:574 Promise._settlePromise
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:574:26
  
  - promise.js:614 Promise._settlePromise0
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:694:18
  

error UNHANDLED REJECTION


  Error: Couldn't find the required `xsel` binary. On Debian/Ubuntu you can inst  all it with: sudo apt install xsel
  
  - envinfo.js:1 i
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:65651
  
  - envinfo.js:1 Object.copySync
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:66177
  
  - envinfo.js:1 Object.t.writeSync.e [as writeSync]
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:48777
  
  - envinfo.js:1 
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:46934
  
  - envinfo.js:1 Promise.all.then.e
    [lib]/[gatsby-cli]/[envinfo]/dist/envinfo.js:1:46949
  
  - util.js:16 tryCatcher
    [lib]/[gatsby-cli]/[bluebird]/js/release/util.js:16:23
  
  - promise.js:512 Promise._settlePromiseFromHandler
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:512:31
  
  - promise.js:569 Promise._settlePromise
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:569:18
  
  - promise.js:614 Promise._settlePromise0
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:694:18
  
  - promise.js:638 Promise._fulfill
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:638:18
  
  - promise_array.js:126 PromiseArray._resolve
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise_array.js:126:19
  
  - promise_array.js:144 PromiseArray._promiseFulfilled
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise_array.js:144:14
  
  - promise.js:574 Promise._settlePromise
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:574:26
  
  - promise.js:614 Promise._settlePromise0
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:614:10
  
  - promise.js:694 Promise._settlePromises
    [lib]/[gatsby-cli]/[bluebird]/js/release/promise.js:694:18
```

**Comment by MarkyC** [comment]
I think this is due to using node v8. Using node v10 solves this problem

**Comment by piotrkwiecinski** [comment]
There is a fix for the issue in gatsby@2.0.75
Please try updating the package and the incompatibility should be resolved.

**Comment by omrllm** [comment]
> There is a fix for the issue in gatsby@2.0.75
> Please try updating the package and the incompatibility should be resolved.

I have gatsby -v 2.4.7, so I should downgrade or what?

**Comment by omrllm** [comment]
You are correct! 

just `npm update` inside your website folder and you are done :smile: 

Thank you! @piotrkwiecinski
